temporal-sdk/src/main/java/io/temporal/internal/sync/SyncWorkflowContext.java

```diff
@@ -139,15 +141,29 @@ public DeterministicRunner getRunner() {
     return runner;
   }
 
-  public WorkflowOutboundCallsInterceptor getWorkflowInterceptor() {
+  public WorkflowOutboundCallsInterceptor getWorkflowOutboundInterceptor() {
     // This is needed for unit tests that create DeterministicRunner directly.
-    return headInterceptor == null ? this : headInterceptor;
+    return headOutboundInterceptor == null ? this : headOutboundInterceptor;
   }
 
-  public void setHeadInterceptor(WorkflowOutboundCallsInterceptor head) {
-    if (headInterceptor == null) {
-      runner.setInterceptorHead(head);
-      this.headInterceptor = head;
+  public WorkflowInboundCallsInterceptor getWorkflowInboundInterceptor() {
+    // This is needed for unit tests that create DeterministicRunner directly.
+    return headInboundInterceptor == null
+        ? new BaseRootWorkflowInboundCallsInterceptor(this)
+        : headInboundInterceptor;
+  }
+
+  public void initHeadOutboundCallsInterceptor(WorkflowOutboundCallsInterceptor head) {
+    if (headOutboundInterceptor == null) {
```

Comment:
Why this check?

Reply:
changed to a throwing an exception. see the explanation behind.